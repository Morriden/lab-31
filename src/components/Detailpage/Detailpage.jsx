import React from 'react';
import PropTypes from 'prop-types';

const Detailpage = ({ name, status, species, image }) => (
  <section>
    <div>{name}</div>
    <div>{status}</div>
    <div>{species}</div>
    <img src={image} />
  </section>
);

Detailpage.propTypes = {
  name: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Detailpage;
