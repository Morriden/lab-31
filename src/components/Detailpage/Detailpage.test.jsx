import React from 'react';
import { shallow } from 'enzyme';
import Detailpage from './Detailpage';


describe('detailpage component', () => {
  it('should show a name status species and image', () => {
    const wrapper = shallow(
      <Detailpage 
        name='test name' 
        status='test status' 
        species='test species' 
        image='test image'
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
