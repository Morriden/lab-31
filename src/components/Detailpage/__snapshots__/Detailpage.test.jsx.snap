// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detailpage component should show a name status species and image 1`] = `
<section>
  <div>
    test name
  </div>
  ,
  <div>
    test status
  </div>
  ,
  <div>
    test species
  </div>
  ,
  <img
    src="test image"
  />
</section>
`;
